### ISSUE TITLE
Insertion Session for MinHashLSHEnsemble

### ISSUE THREAD

**Issue by achachet** [ekzhu/datasketch#115]
Hi, 

I've noticed in the doc that it is possible to create a MinHashLSH index iteratively with the insert() method. However, MinHashLSHEnsemble only has an index() method that requires to have all the minhashes loaded in memory at once. 

It is quite problematic for my use case given the size of the dataset I am working on (billions of entries) 

Is there a structural reason as to why this cannot be done, or is it maybe a feature that you plan to develop in the future?

Thank you so much for sharing this great package.

**Comment by ekzhu** [comment]
Yes. Even though you can pass an iterator as `entries` in the index() method, the implementation of index() still loads all entries in memory. This is because it needs to have a global view of the inserted set sizes to create partitions. 

You can still use `MinHashLSH` for containment, with some manual tuning of the `threshold` parameter. Because Jaccard is always the lower-bound of the corresponding containment, i.e., `jaccard(x, y) <= |x \intersect y| / |x|`, you can use a MinHashLSH with `threshold` set to your desired containment threshold, and tune your way up to find the best `threshold` for your dataset.
